keps/sig-storage/625-csi-migration/README.md

```diff
@@ -156,10 +165,155 @@ Running this set of tests is optional for a per-provider basis. We would
+
+* **What specific metrics should inform a rollback?**
+  We have metrics on the CSI sidecar side called `csi_operation_duration_seconds` and core k8s metrics on both kube-controller-manager and kubelet side called `storage_operation_duration_seconds`. Both of them will have a `migrated` field to whether indicate this operation is a migrated PV operation. If the error ratio of such PV has an unusual strike or have an overall high error ratio, it means something goes wrong and we need to rollback.
+
+* **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
+To turn it on by default in Beta, we require each in-tree plugin to at least manually test the upgrade->downgrade->upgrade path.
+For GA, we require such test exists in each driver's test CI.
+
+* **Is the rollout accompanied by any deprecations and/or removals of features, APIs,
+fields of API types, flags, etc.?**
+There will not be API removal in CSI migration itself. But eventually when CSI migration is all finished. We will plan to remove all in-tree plugins. So deprecation and code-removal will be required eventually.
+
+### Monitoring Requirements
+
+* **How can an operator determine if the feature is in use by workloads?**
+  We will have metrics `csi_sidecar_duration_seconds` on the CSI sidecars and `storage_operation_duration_seconds` on the kube-controller-manager and kubelet side to indicate whether this operation is a migrated operation or not.
+
+* **What are the SLIs (Service Level Indicators) an operator can use to determine
+the health of the service?**
+  - [x] Metrics
+    - Metric name: csi_sidecar_duration_seconds && storage_operation_duration_seconds
+    - [Optional] Aggregation method:
+    - Components exposing the metric: CSI sidecars, kubelet, kube-controller-manager
+  - [x] Other (treat as last resort)
+    - Details: Pod using in-tree plugin has failure.
+
+* **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
+  - per-day percentage of API calls using in-tree plugin that enables the CSI migration compared to not enable the feature has no more than 3% error ratio.
+
+* **Are there any missing metrics that would be useful to have to improve observability
+of this feature?**
+Node side CSI operation metrics. It will be implemented in the GA phase.
+
+### Dependencies
+
+
+* **Does this feature depend on any specific services running in the cluster?**
+
+  - Corresponding CSI Driver for in-tree CSI migration enabled plugin
+    - Usage description:
+      - Impact of its outage on the feature: in-tree plugin stops working without the CSI Driver properly setup
+      - Impact of its degraded performance or high-error rates on the feature: Error or performance decrease will be reflected for volume intensive operation
+
+
+### Scalability
+
+* **Will enabling / using this feature result in any new API calls?**
+  No
```

Comment:
In the question about object count/size you're mentioning CSI driver to be installed.

Obviously these drives send API calls, so we might want to mention that here too for completeness.

Reply:
Good catch, thanks!